authors/apps/authentication/views.py

```diff
@@ -105,3 +112,74 @@ def update(self, request, *args, **kwargs):
         serializer.save()
 
         return Response(serializer.data, status=status.HTTP_200_OK)
+
+class PasswordResetView(APIView):
+
+    def post(self, request):
+        if 'email' not in request.data:
+            raise exceptions.ValidationError("Please provide an Email Address")
+        email=request.data["email"]
+        try:
+            if not email:
+                raise exceptions.ValidationError("Please fill in the email field")
+
+            if not re.match(r"(^[a-zA-Z0-9_.+-]+@[a-zA-Z0-9-]+\.[a-zA-Z0-9-.]+$)", email):
+                raise exceptions.ValidationError("Invalid Email Address")
+
+            if not User.objects.filter(email = email).first():
+                raise exceptions.ValidationError("This Email Address is not attached to any account")
+
+            serializer = URLSafeTimedSerializer(os.environ.get("SECRET_KEY"))
+            token = serializer.dumps(email, salt=os.environ.get("SECURITY_PASSWORD_SALT"))
+            
+            BASE_URL = request.get_host()
+            url_scheme = request.scheme
+            reset_link = "{}://{}/api/users/change-password/{}/".format(url_scheme,BASE_URL,token)
+            recipient = [email]
+            sender = os.getenv('EMAIL_HOST_USER')
+            subject = 'Author\'s Haven Password Reset'
+            body = "You requested to change your account password.\n\
+Click on the link below to complete changing your password.\n\n{}\n\
+Ignore and Delete this email if you did not make this request.\n\n\t\
+Author\'s Haven by The Realers.".format(reset_link)
+            
+            send_mail(subject, body, sender, recipient, fail_silently=True)
+
+            data = {
+                "message": "Please check your email inbox for the Password Reset link we've sent",
+                "status": status.HTTP_200_OK
+            }
+            return Response(data, status=status.HTTP_200_OK)
+        except KeyError:
+            raise exceptions.ValidationError("Please provide a valid Email Address")
+
+class CreateNewPasswordView(APIView):
```

Comment:
Please look into these code climate issue @HabibSentongo

Reply:
This was successfully resolved